### PR TITLE
MM-39711 - Gekidou Settings Push Notification functionality

### PR DESCRIPTION
**Pull request by avinashlng1080**
#### Summary
Adds the push notification settings.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-39711

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/screens/settings/notification_push/notification_push.tsx [hunk 1]
```diff
@@ -27,31 +36,90 @@ const getStyleSheet = makeStyleSheetFromTheme((theme) => {
...
+    }, [pushSend]);
 
-    const onMobilePushThreadChanged = (status: PushStatus) => {
-        setPushThreadPref(status);
-    };
+    const onMobilePushThreadChanged = useCallback(() => {
+        setPushThreadPref(pushThread === 'all' ? 'mention' : 'all');
+    }, [pushThread]);
+
+    const canSave = useCallback(() => {
```
Comment (enahum) [inline app/screens/settings/notification_push/notification_push.tsx:75]: same as the other PR's

File: app/screens/settings/notification_push/notification_push.tsx [hunk 2]
```diff
@@ -27,31 +36,90 @@ const getStyleSheet = makeStyleSheetFromTheme((theme) => {
...
+            id: SAVE_NOTIF_BUTTON_ID,
+            enabled: false,
+            showAsAction: 'always' as const,
+            testID: 'notification_settings.save.button',
+            color: theme.sidebarHeaderTextColor,
+            text: intl.formatMessage({id: 'settings.save', defaultMessage: 'Save'}),
+        };
+    }, [theme.sidebarHeaderTextColor]);
+
+    const close = useCallback(() => popTopScreen(componentId), [componentId]);
```
Comment (enahum) [inline app/screens/settings/notification_push/notification_push.tsx:102]: same

File: app/screens/settings/notification_push/push_send.tsx [hunk 3]
```diff
@@ -61,7 +61,7 @@ const MobileSendPush = ({sendPushNotifications, pushStatus, setMobilePushPref}:
             {sendPushNotifications &&
                 <>
                     <OptionItem
-                        action={setMobilePushPref}
+                        action={() => setMobilePushPref('all')}
```
Comment (enahum) [inline app/screens/settings/notification_push/push_send.tsx:64]: is this intended to be an inline function?, same with the others

File: app/screens/settings/notification_push/push_status.tsx [hunk 4]
```diff
@@ -53,7 +53,7 @@ const MobilePushStatus = ({pushStatus, setMobilePushStatus}: MobilePushStatusPro
             headerStyles={styles.upperCase}
         >
             <OptionItem
-                action={setMobilePushStatus}
+                action={() => setMobilePushStatus('online')}
```
Comment (enahum) [inline app/screens/settings/notification_push/push_status.tsx:56]: same question about inline function

**Review by enahum — CHANGES_REQUESTED** [review]
need to run i18n-extract

File: app/screens/settings/notification_push/notification_push.tsx (hunk 2 above)
Comment (avinashlng1080) [reply app/screens/settings/notification_push/notification_push.tsx:86]: corrected

File: app/screens/settings/notification_push/push_send.tsx (hunk 3 above)
Comment (avinashlng1080) [reply app/screens/settings/notification_push/push_send.tsx]: corrected

File: app/screens/settings/notification_push/push_status.tsx (hunk 4 above)
Comment (avinashlng1080) [reply app/screens/settings/notification_push/push_status.tsx]: corrected

File: app/screens/settings/notification_push/notification_push.tsx (hunk 1 above)
Comment (avinashlng1080) [reply app/screens/settings/notification_push/notification_push.tsx]: corrected

**Review by enahum — APPROVED** [review]

**Review by enahum — APPROVED** [review]

**Review by larkox — APPROVED** [review]
LGTM, just one small improvement.

File: app/screens/settings/notification_push/notification_push.tsx [hunk 5]
```diff
@@ -27,31 +36,86 @@ const getStyleSheet = makeStyleSheetFromTheme((theme) => {
...
         setPushStatus(status);
-    };
+    }, []);
 
-    const setMobilePushPref = (status: PushStatus) => {
-        setPushPref(status);
-    };
+    const setMobilePushPref = useCallback((status: PushStatus) => {
+        setPushSend(status);
+    }, [pushSend]);
```
[the remark this answers is not in the document]
Comment (avinashlng1080) [reply app/screens/settings/notification_push/notification_push.tsx]: implemented

**Pull request merged**